convert2rhel/actions/conversion/preserve_only_rhel_kernel.py

```diff
@@ -0,0 +1,207 @@
+# Copyright(C) 2024 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+import glob
+import logging
+import os
+import re
+
+from convert2rhel import actions, exceptions, pkghandler, pkgmanager, utils
+from convert2rhel.systeminfo import system_info
+
+
+loggerinst = logging.getLogger(__name__)
+
+# create classes for installing, removing non rhel kernels, install additional rhel kernel pkgs and update rhel kernel if needed
```

Comment:
Turn this in a docstring for the class

Reply:
Sorry that was just a leftover comment for planning the classes